### PR TITLE
#57 Refactor and show profile on homepage after logging in

### PR DESCRIPTION
**Pull request by KhangZhengNg**
#57

Now the user's profile with profile picture, last name, first name, telephone number, and email is being displayed on the homepage. Also added AppBarLayout to the homepage so that we can have overflow menus.

### CODE REVIEW

**Review by refatK — APPROVED** [review]
Tried the app and worked as expected, great job:
image

Everything else seems to work like before, and code looks good as well.

File: app/build.gradle [hunk 1]
```diff
@@ -58,6 +58,8 @@ dependencies {
     implementation 'androidx.navigation:navigation-fragment:2.3.0'
     implementation 'androidx.navigation:navigation-ui:2.3.0'
     implementation 'de.hdodenhof:circleimageview:3.1.0'
+    implementation 'androidx.wear:wear:1.0.0'
+    implementation 'com.squareup.picasso:picasso:2.71828'
```
[the remark this answers is not in the document]
Comment (KhangZhengNg) [reply app/build.gradle]: The library wear is automatically generated when I ran and installed the app on my phone. It is meant for Wear OS by Google smartwatches. I removed it in following commit. Picasso is a powerful image downloading and caching library for Android. It takes image url and convert it into bitmap.

File: app/src/main/java/com/example/bigowlapp/activity/HomePageActivity.java [hunk 2]
```diff
@@ -37,65 +47,82 @@ protected void onStart() {
     }
 
     protected void initialize() {
+        scrollView = findViewById(R.id.scrollView);
+        scrollView.setVisibility(View.GONE);
         try {
+            if (FirebaseAuth.getInstance().getCurrentUser() != null) {
+                String currentUserUid = FirebaseAuth.getInstance().getCurrentUser().getUid();
+                DocumentReference docRef = db.collection("users").document(currentUserUid);
```
[the remark this answers is not in the document]
Comment (KhangZhengNg) [reply app/src/main/java/com/example/bigowlapp/activity/HomePageActivity.java]: thanks and it's done in new commits.

File: app/src/main/java/com/example/bigowlapp/activity/EditProfileActivity.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_edit_profile);
+        initialize();
+    }
+
+    protected void initialize() {
+        editProfileViewModel = new ViewModelProvider(this).get(EditProfileViewModel.class);
+
+        edit_buttonCancel = findViewById(R.id.edit_buttonCancel);
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/activity/EditProfileActivity.java:29]: Please use camelcase (no underscores) for names of java objects

**Review by refatK — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/activity_edit_profile.xml [hunk 4]
```diff
@@ -6,29 +6,29 @@
     android:layout_height="match_parent">
 
     <Button
-        android:id="@+id/edit_buttonCancel"
+        android:id="@+id/editButtonCancel"
```
Comment (refatK) [inline app/src/main/res/layout/activity_edit_profile.xml:9]: For xml, the convention actually is underscores, so please change it back.

Java = camelCase
Xml= underscores

**Review by refatK — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/activity_edit_profile.xml (hunk 4 above)
Comment (KhangZhengNg) [reply app/src/main/res/layout/activity_edit_profile.xml:9]: ok got it. Thanks for point it out!

**Review by refatK — APPROVED** [review]
Still good after the further changes!

**Review by Landry333 — CHANGES_REQUESTED** [review]
The application crashes when one try to sign up without a name. Can you fix that or just make the name compulsory for signing up like the phone number?

**Review by darkheros12 — CHANGES_REQUESTED** [review]
I actually like that we have a profile for user now. However, I would request that we have the actual UI that @SeifElDerby did for us to avoid going back to the file and do it all over again. It does not take that long to do.

**Comment by Landry333** [comment]
For now we don't have the features on the UI Seif did. Let's add this PR to our work as other members were waiting for it. We will still be able to modify the UI later on

**Review by Landry333 — APPROVED** [review]
The requested changes were made. The rest of the PR seems ok to me.

**Review by darkheros12 — APPROVED** [review]

**Comment by darkheros12** [comment]
Please pull from staging before PR

**Review by darkheros12 — CHANGES_REQUESTED** [review]
please pull from staging before the PR. Once it is done I will approve it.

**Comment by Landry333** [comment]
The branch has been updated with staging

**Review by darkheros12 — APPROVED** [review]

**Pull request merged**